jax/_src/lax/control_flow/for_loop.py

```diff
@@ -132,7 +133,9 @@ def wrapped_body(i, refs):
   nsteps, = nsteps
   flat_state, state_tree = tree_flatten(init_state)
   state_avals = map(state_utils.val_to_ref_aval, flat_state)
-  idx_aval = core.ShapedArray((), jnp.dtype("int32"))
+  idx_aval = core.ShapedArray(
+      (), jnp.int64 if config.enable_x64.value else jnp.int32
```

Comment:
You could use `dtypes.canonicalize_dtype` which contains that conditional (and is actually more likely to work when we use the x64 context manager)

Reply:
Done.